gateway/implementation/src/main/java/eu/openg/guessnumberapi/gateway/implementation/PostgreSQL/PostgresqlGameRepo.java

```diff
@@ -0,0 +1,96 @@
+        connection = postgresqlConnection.connectToPostgresqlDatabase();
+        createGameTableIfNotExists();
+    }
+
+    private void createGameTableIfNotExists() {
+        try (Statement statement = connection.createStatement()) {
+            statement.executeUpdate(QueryUtils.CREATE_TABLE_QUERY);
+            LOGGER.info("PostgreSql database table created successfully");
+        } catch (SQLException e) {
+            throw logErrorAndReturnNewException("CREATE TABLE failed. game table could not be created.", e);
+        }
+    }
+
+    @Override
+    public int saveNewGameAndReturnId(Game game) {
+        try (PreparedStatement statement = connection.prepareStatement(QueryUtils.INSERT_GAME_QUERY)) {
+            int id = insertAndReturnId(statement, game);
+            LOGGER.info("PostgreSql new game insert successful.");
+            return id;
+        } catch (SQLException e) {
+            throw logErrorAndReturnNewException("INSERT failed. New game could not be inserted.", e);
+        }
+    }
+
+    private int insertAndReturnId(PreparedStatement statement, Game game) throws SQLException {
+        statement.setInt(1, game.getGuessCount());
+        statement.setInt(2, game.getActualNumber());
+        return executeQueryAndReturnField(statement, "GAMEID");
+    }
+
+    @Override
+    public int incrementAndReturnGuessCount(int gameId) {
+        try (PreparedStatement statement = connection.prepareStatement(QueryUtils.UPDATE_GUESSCOUNT_QUERY)) {
+            int updatedGuessCount = updateAndReturnGuessCount(statement, gameId);
+            LOGGER.info("PostgreSql guessCount increment update successful.");
+            return updatedGuessCount;
+        } catch (SQLException e) {
+            throw logErrorAndReturnNewException("UPDATE failed. guessCount could not be incremented", e);
+        }
+    }
+
+    private int updateAndReturnGuessCount(PreparedStatement statement, int gameId) throws SQLException {
+        statement.setInt(1, gameId);
+        return executeQueryAndReturnField(statement, "GUESSCOUNT");
+    }
+
+    private int executeQueryAndReturnField(PreparedStatement statement, String fieldToReturn) throws SQLException {
+        try (ResultSet rs = statement.executeQuery()) {
+            if (rs.next())
+                return rs.getInt(fieldToReturn);
+        }
+        throw new PostgresqlException("Failed. Statement did not return any results.");
+    }
+
+    @Override
+    public Game fetchGame(int gameId) {
+        try (PreparedStatement statement = connection.prepareStatement(QueryUtils.SELECT_GAME_QUERY)) {
+            statement.setInt(1, gameId);
+            return selectAndReturnGameFromDB(statement);
+        } catch (SQLException e) {
+            throw logErrorAndReturnNewException("SELECT failed. Game could not be fetched from database.", e);
+        }
+    }
+
+    private Game selectAndReturnGameFromDB(PreparedStatement statement) throws SQLException {
```

Comment:
nieko jis neselectina, net nežino apie querio sintaksę, tiesiog gauna statementą, kurį executina ir td išrenka duomenis. taip pat siūlyčiau apie solid dar biški paskaityt, jeigu jau metodo name yra "and" tai turbūt single responsibility net nekvepia